### PR TITLE
Tickets/dm33154: State Transition tests

### PR DESCRIPTION
**Pull request by rbovill**

### CODE REVIEW

File: python/lsst/ts/IntegrationTests/obssys2_standby_disabled.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["ObsSys2StandbyDisabled"]
+
+from lsst.ts.IntegrationTests import BaseScript
+from .config_registry import registry
+
+
+class ObsSys2StandbyDisabled(BaseScript):
+    """Execute the given Main Telescope Standard or External
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/obssys2_standby_disabled.py:28]: Just remove Main Telescope from this comment as it doesn't really apply.

File: python/lsst/ts/IntegrationTests/obssys2_state_transition_configs.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+from .config_registry import registry
+
+
+# Add the State Transition script configurations to the registry.
+
+# obssys2_standby_disabled
+yaml_string = yaml.safe_load(
+    """
+    data:
+    - - Scheduler:1
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/obssys2_state_transition_configs.py:33]: Schedulers have configuration that needs to be applied. The Confluence page has been updated.

File: python/lsst/ts/IntegrationTests/obssys2_disabled_enabled.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["ObsSys2DisabledEnabled"]
+
+from lsst.ts.IntegrationTests import BaseScript
+from .config_registry import registry
+
+
+class ObsSys2DisabledEnabled(BaseScript):
+    """Execute the given Main Telescope Standard or External
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/obssys2_disabled_enabled.py:28]: Just remove Main Telescope from this comment as it doesn't really apply.

File: python/lsst/ts/IntegrationTests/mt_state_transition_configs.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
+# This file is part of ts_IntegrationTests.
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/mt_state_transition_configs.py:1]: I'd call this maintel_state.. for clarity.

File: python/lsst/ts/IntegrationTests/at_state_transition_configs.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
+# This file is part of ts_IntegrationTests.
```
Comment (mareuter) [inline python/lsst/ts/IntegrationTests/at_state_transition_configs.py:1]: I'd call this auxtel_state... for clarity.

**Review by mareuter — CHANGES_REQUESTED** [review]
Cleanup work to do. We also need to add some EAS related state transition scripts. The Confluence page has been updated with the necessary information.

File: python/lsst/ts/IntegrationTests/at_state_transition_configs.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+registry["auxtel_camera_disabled_enabled"] = yaml.safe_dump(
+    yaml_string, explicit_start=True, canonical=True
+)
+
+# auxtel_offline_standby
+yaml_string = yaml.safe_load(
+    """
+    data:
+    - - ATCamera
+    - DISABLED
```
[the remark this answers is not in the document]
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/at_state_transition_configs.py:127]: Fixed

File: python/lsst/ts/IntegrationTests/aux_tel_disabled_enabled.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
+# This file is part of ts_IntegrationTests
```
[the remark this answers is not in the document]
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/aux_tel_disabled_enabled.py:1]: fixed

File: python/lsst/ts/IntegrationTests/mt_state_transition_configs.py [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+
+# Add the State Transition script configurations to the registry.
+
+# maintel_offline_standby
+yaml_string = yaml.safe_load(
+    """
+    data:
+    - - CCCamera
+    - DISABLED
```
[the remark this answers is not in the document]
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/mt_state_transition_configs.py:34]: Fixed

File: python/lsst/ts/IntegrationTests/obssys2_standby_disabled.py (hunk 1 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/obssys2_standby_disabled.py:28]: Fixed

File: python/lsst/ts/IntegrationTests/obssys2_state_transition_configs.py (hunk 2 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/obssys2_state_transition_configs.py:33]: Fixed

File: python/lsst/ts/IntegrationTests/obssys2_disabled_enabled.py (hunk 3 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/obssys2_disabled_enabled.py:28]: Fixed

File: python/lsst/ts/IntegrationTests/at_state_transition_configs.py (hunk 5 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/at_state_transition_configs.py:1]: Fixed

File: python/lsst/ts/IntegrationTests/mt_state_transition_configs.py (hunk 4 above)
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/mt_state_transition_configs.py:1]: Fixed

File: python/lsst/ts/IntegrationTests/enabled_offline_state_transition_configs.py [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    - - OCPS:2
+    - OFFLINE
+    """
+)
+
+registry["sched_ocps_enabled_offline"] = yaml.safe_dump(
+    yaml_string, explicit_start=True, canonical=True
+)
+
+# dsm_enabled_offline
```
[the remark this answers is not in the document]
Comment (rbovill) [reply python/lsst/ts/IntegrationTests/enabled_offline_state_transition_configs.py:48]: Fixed

File: tests/test_auxtel_state_transitions.py [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        # execute the scripts.
+        script_class = AuxTelDisabledEnabled()
+        await script_class.run()
+        # Get number of scripts
+        num_scripts = len(script_class.scripts)
+        print(f"AuxTel Disabled to Enabled; running {num_scripts} scripts")
+        # Assert script was added to ScriptQueue.
+        self.assertEqual(len(self.controller.queue_list), num_scripts)
+
+    async def test_enabled_offline(self):
```
[the remark this answers is not in the document]
Comment (rbovill) [reply tests/test_auxtel_state_transitions.py]: Fixed

File: tests/test_enabled_offline.py [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+
+from lsst.ts import salobj
+from lsst.ts.IntegrationTests import ScriptQueueController
+from lsst.ts.IntegrationTests import EnabledOffline
+
+
+class AuxTelStateTransitionTestCase(unittest.IsolatedAsyncioTestCase):
```
Comment (mareuter) [inline tests/test_enabled_offline.py:31]: Need a different test class name here. EnabledOfflineTestCase, EnabledOfflineStateTransitionTestCase or ShutdownStateTransitionTestCase work.

**Review by mareuter — APPROVED** [review]
Need to fix one test class name, but otherwise this looks great. Thanks for all the changes.

File: tests/test_enabled_offline.py (hunk 11 above)
Comment (rbovill) [reply tests/test_enabled_offline.py]: Good catch.  Fixed.

**Pull request merged**